### ISSUE TITLE
How to get y_pred and y_true for two classes using the inference results

### ISSUE THREAD

**Issue by jpainam** [facebookresearch/maskrcnn-benchmark#1300]
## ❓ Questions and Help
Hi, I training maskrcnn on a custom dataset with two classes( background excluded). After testing, I get some files `segm.json, predictions.pth, coco_results.pth,  bbox.json` in the inference folder. 
I want to get the two sets `y_true` and `y_pred` after running my tests. Which files or how can I get these sets without running my test on each image? Thank you.

**Comment by Eldad27** [comment]
@jpainam I found a partial description that could help understand the contents of the inference folder
Those files are intermediate results of the inference, and they contain different information. 

a.	'coco_results.pth ' is just contains the mIoU, 
b.	 'prediction.pth ' holds the outputs of the model for every image in the dataset
c.	the masks are stored in  ' prediction.pth '
d.	the rle encodings are available in the  'python segm.json file, '

However, I'm not sure how to extract  ' y_pred ' and ' y_true' yet. Have you been able to resolve this issue?

**Comment by jpainam** [comment]
Hi. thank you for pointing out to the `predictions.pth` file. The file contains a list of `BoxList` object defined in https://github.com/facebookresearch/maskrcnn-benchmark/blob/57eec25b75144d9fb1a6857f32553e1574177daf/maskrcnn_benchmark/structures/bounding_box.py#L9
`y_pred` is saved as `.extra_fields` .
Anyone interested can use this piece of code to read its predictions
```python
predictions = torch.load("predictions.pth")
y_preds = []
for box in predictions:
   scores = box.get_field("scores")
   labels = box.get_field("labels")
   mask = box.get_field("mask")
   mask_scores = box.get_field("mask_scores")
   y_preds = np.concatenate((y_preds, labels.numpy()), axis=None)
y_preds = np.array(y_preds, dtype=np.int32)
```

**Comment by Eldad27** [comment]
@jpainam Thanks very much. Well appreciated.

**Comment by Eldad27** [comment]
I'm also dealing with binary classification (classes = 1 and 2). I have successfully extracted `y_pred` from `predictions.pth` inference file using the code below as suggested.

```
import torch
import numpy as np
predictions = torch.load("predictions.pth")
y_preds = []
for box in predictions:
   scores = box.get_field("scores")
   labels = box.get_field("labels")
   mask = box.get_field("mask")
   mask_scores = box.get_field("mask_scores")
   y_preds = np.concatenate((y_preds, labels.numpy()), axis=None)
y_preds = np.array(y_preds, dtype=np.int32)
print(y_pred.shape)
print(y_preds)
```
I get an output for `y_pred`  similar to output below. However, this does not add up to `y_true.` (25 objects in total),   Meaning some detections (about 8) were missed.

```
(17)
[1 1 1 2 1 2 1 1 1 1 2 1 1 1 1 2 1 2]
```  

How do I extract the 8 object labels for the classes that weren't detected? How do I know the class(es) that were not detected?

Please help!
